controllers/chat/group/removeGroupMember.js

```diff
@@ -16,6 +16,16 @@ const removeGroupMember = async (req, res) => {
   const targetUserModel = await UsersFunction.findUserById(User, targetUserId);
 
   try {
+    const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    const currentChannel = await client.channel(CHANNEL_TYPE_STRING.GROUP, channelId);
+    const queriedChannel = await client.queryChannels({
+      type: CHANNEL_TYPE_STRING.GROUP,
+      id: channelId,
+      members: {$in: [userId]}
+    });
+
+    let all_members = queriedChannel[0].data.better_channel_member.map((member) => member.user.id);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a potential performance issue. The `queryChannels` function is used to fetch the channel and its members, but this operation seems unnecessary because you're already fetching the channel with `client.channel(CHANNEL_TYPE_STRING.GROUP, channelId)`. If the purpose of this query is to get all members of the channel, consider using the `channel.query()` method on the `currentChannel` object instead of making another API call.

```diff
-    const queriedChannel = await client.queryChannels({
-      type: CHANNEL_TYPE_STRING.GROUP,
-      id: channelId,
-      members: {$in: [userId]}
-    });
-
-    let all_members = queriedChannel[0].data.better_channel_member.map((member) => member.user.id);
+    const {channel} = await currentChannel.query();
+    let all_members = channel.members.map((member) => member.user.id);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->